adam/language_specific/chinese/chinese_syntax.py

```diff
@@ -0,0 +1,74 @@
+"""Current implementation of Chinese syntax file; has been checked
+by native speaker asked for grammaticality judgments."""
+
+from typing import Tuple
+from immutablecollections import ImmutableDict, immutabledict
+from adam.language.dependency import (
+    DependencyRole,
+    HEAD,
+    PartOfSpeechTag,
+    RoleOrderDependencyTreeLinearizer,
+)
+from adam.language.dependency.universal_dependencies import (
+    ADJECTIVAL_MODIFIER,
+    ADVERBIAL_MODIFIER,
+    CASE_POSSESSIVE,
+    CASE_SPATIAL,
+    INDIRECT_OBJECT,
+    NOMINAL_MODIFIER,
+    NOMINAL_MODIFIER_POSSESSIVE,
+    NOMINAL_SUBJECT,
+    NOUN,
+    NUMERIC_MODIFIER,
+    OBJECT,
+    OBLIQUE_NOMINAL,
+    PROPER_NOUN,
+    VERB,
+    IS_ATTRIBUTE,
+    CLASSIFIER,
+)
+
+_CHINESE_HEAD_TO_ROLE_ORDER: ImmutableDict[
+    PartOfSpeechTag, Tuple[DependencyRole, ...]
+] = [
+    # Currently we treat locations associated with verbs as IO's since obliques occur before the verb.
+    # For example, "on the table" is preverbal in "I eat on the table" but post-verbal in "move the book to the table"
+    # TODO: find a better way to handle the above
```

Comment:
Can you add a link to the relevant GitHub issue for this comment? Marked with 🔗 in the future.